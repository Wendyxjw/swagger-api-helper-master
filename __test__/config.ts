import { Url } from '../src/generate/interfaces';

export const Authorization = '';

export const urls: Url[] = [
    ['https://petstore.swagger.io/v2/swagger.json', 'swaggerDirname'],
    [`https://petstore.swagger.io/v2/swagger.json`, 'swaggerDirname2'],
];
