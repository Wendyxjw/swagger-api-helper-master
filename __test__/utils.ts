import fs from 'fs';
import { join } from 'path';

// 删除文件夹
export function removeFolder(path: string) {
    if (fs.existsSync(path)) {
        const files = fs.readdirSync(path);
        files.forEach(file => {
            const curPath = join(path, file);
            if (fs.statSync(curPath).isDirectory()) {
                removeFolder(curPath);
            } else {
                fs.unlinkSync(curPath);
            }
        });
        fs.rmdirSync(path);
    }
}
