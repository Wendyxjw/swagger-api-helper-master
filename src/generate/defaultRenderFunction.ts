import path from 'path';
import { RenderFunctionOptions } from './interfaces';

export default ({
    basePath,
    url,
    name,
    method,
    hasBody,
    hasParams,
    hasQuery,
    responseType,
    payloadType,
    queryType,
    paramsTypeMap,
    extraFetchOptionsParaName,
}: RenderFunctionOptions): string => {
    const fullUrl = path.join(basePath, url);
    const paramsString = renderParams({
        hasBody,
        hasParams,
        hasQuery,
        paramsTypeMap,
        extraFetchOptionsParaName,
    });
    return `
export async function ${name}(${renderArgs(payloadType)}) {${paramsString}
  return request<${responseType}>(\`${fullUrl}${hasQuery ? '?${stringify(query)}' : ''}\`, {${
    paramsString ? `\n\t\t...${extraFetchOptionsParaName},` : ''
  }
    method: '${method}',${hasBody ? '\n\t\tbody,' : ''}
  });
}
`;
};

/**
 * 返回方法参数  如 payload: Payload
 * @param {string} interfaces
 * @param {object} payloadInterfaceName
 */
function renderArgs(payloadType?: string) {
    if (payloadType) {
        return `payload: ${payloadType}`;
    }
    return '';
}

/**
 * 返回 方法中的参数内容  如  const { body, query, ...extraFetchOptions } = payload;
 * @param {object} item
 */
function renderParams({
    hasBody,
    hasQuery,
    hasParams,
    paramsTypeMap,
    extraFetchOptionsParaName,
}: {
    hasBody: boolean;
    hasQuery: boolean;
    hasParams: boolean;
    paramsTypeMap: { [name: string]: string };
    extraFetchOptionsParaName: string;
}) {
    const allPara = [
        ...(hasParams ? Object.keys(paramsTypeMap) : []),
        ...(hasBody ? ['body'] : []),
        ...(hasQuery ? ['query'] : []),
    ];
    const paramsString = `{ ${allPara.join(', ')}, ...${extraFetchOptionsParaName} }`;
    return `\n\tconst ${
        allPara.length === 0 ? extraFetchOptionsParaName : paramsString
    } = payload;`;
}
