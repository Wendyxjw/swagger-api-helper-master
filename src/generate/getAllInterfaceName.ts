import { Definitions } from '../interfaces';
import renderRefModelTitle from './getRefModelTitle';

/**
 * 获取definitions的key
 */
export default (definitions: Definitions): string[] => {
    const interfaceNames:string[]=[]
    const definitionKeys = Object.keys(definitions);
    definitionKeys.forEach(key => {
        const interfaceName = renderRefModelTitle({
            ...definitions[key],
            title: key,
        });
        interfaceNames.push(interfaceName)
    });
    return interfaceNames;
};
