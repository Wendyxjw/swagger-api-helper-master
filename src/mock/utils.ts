/**
 * 判断参数名称是不是带指定后缀
 * @param keyName
 * @param likePropNames
 */
export const isPropNameLike = (keyName: string, likePropNames: string[]) => {
    return likePropNames.some((name) => keyName.endsWith(name));
};
