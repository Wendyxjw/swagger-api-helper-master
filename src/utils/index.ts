export { default as fetchSwaggerJson } from './fetchSwaggerJson';
export { default as writeFile } from './writeFile';
export { default as firstCharUpper } from './firstCharUpper';
export { default as getRef } from './getRef';
