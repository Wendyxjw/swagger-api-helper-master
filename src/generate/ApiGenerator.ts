import { SwaggerResponse, SwaggerFetchOptions } from '../interfaces';
import { fetchSwaggerJson, writeFile } from '../utils';
import parseSwaggerJson from '../parseSwaggerJson';
import { Options, Url, GenMessage, Status, GenMessageWrapper, UrlConfig } from './interfaces';
import { join, basename } from 'path';
import getApiModel from './getApiModel';
import getInterfacesModel from './getInterfacesModel';
import getAllInterfaceName from './getAllInterfaceName';


type IUrl = [string, string, UrlConfig, SwaggerFetchOptions | undefined];

// 生成 存放 swagger model 的文件名
const interfaceModelsName = 'interfaces';

class ApiGenerator {
    private options: Options;
    private urls: IUrl[];
    private swaggerResponses: SwaggerResponse[] = [];

    constructor(options: Options) {
        this.options = options;
        this.urls = this.parseUrls(options.urls, options.fetchOptions);
    }

    private parseUrls(urls: Url[], fetchOptions?: SwaggerFetchOptions): IUrl[] {
        return urls.map((url, index) => {
            if (typeof url === 'string') {
                return [url, `swaggerApi${index}`, {}, fetchOptions];
            }
            if (url[2]) {
                return [...url, fetchOptions] as IUrl;
            }
            return url[3] ? (url as IUrl) : [url[0], url[1], url[2], fetchOptions];
        });
    }

    /**
     * 请求 swagger 数据
     */
    async fetch() {
        const fetchApis = this.urls.map(([url, dirname, config, currentFetchOptions]) =>
            fetchSwaggerJson(url, currentFetchOptions)
        );
        this.swaggerResponses = await Promise.all(fetchApis);
        return this.swaggerResponses;
    }

    /**
     * 一般在等待 this.fetch 完成后再执行，否则不会生成
     * 生成接口文件
     */
    async generate() {
        const { tagAlias = {}, outputPath, include = [], exclude = [], notRequiredInterfaces } = this.options;
        const messages: GenMessageWrapper[] = [];
        let curIndex = 0;
        for await (const swaggerResponse of this.swaggerResponses) {
            const dirname = this.urls[curIndex][1];
            const urlConfig = this.urls[curIndex][2];

            const { swaggerObj, basePath, definitions } = parseSwaggerJson(swaggerResponse);
            const allInterfaceName = getAllInterfaceName(definitions)
            const includeKeys = Object.keys(swaggerObj)
                .map(key => key.trim())
                .filter(key => swaggerObjFilter(key, [...include, ...(urlConfig?.include || [])], [...exclude, ...(urlConfig?.exclude || [])]));
            const { apiModelPromises, globalInterfaceNamesSet } = includeKeys.reduce<{
                apiModelPromises: Promise<GenMessage>[];
                globalInterfaceNamesSet: Set<string>;
            }>(
                (target, key) => {
                    const alias = tagAlias[key];
                    const filename = join(outputPath, `${dirname}/${alias ? alias : key}.ts`);
                    const { content, relatedInterfaceNames } = getApiModel(
                        swaggerObj[key].paths,
                        key,
                        basePath,
                        filename,
                        this.options,
                        dirname,
                        allInterfaceName
                    );
                    target.apiModelPromises.push(genFile(filename, content));
                    relatedInterfaceNames.forEach(name => target.globalInterfaceNamesSet.add(name));
                    return target;
                },
                {
                    apiModelPromises: [],
                    globalInterfaceNamesSet: new Set<string>(),
                }
            );
            // swagger 所有的 定义的模型，生成到一个 interfaces 文件
            const interfaceModelsContent = getInterfacesModel(definitions, [
                ...globalInterfaceNamesSet,
            ], notRequiredInterfaces);
            const interfacesFilename = join(outputPath, `${dirname}/${interfaceModelsName}.ts`);
            const interfacesModelPromise = genFile(interfacesFilename, interfaceModelsContent);

            const allPromises = [...apiModelPromises, interfacesModelPromise];
            const message = await allSettled(allPromises);
            messages.push(message);

            curIndex += 1;
        }

        return messages;
    }
}

export default ApiGenerator;

/**
 * 在 swaggerObj 中筛选出需要生成的 接口
 * @param key 当前的 tag
 * @param include 需要生成的 tag 优先级大于 exclude
 * @param exclude 不需要生成的 tag
 */
function swaggerObjFilter(key: string, include?: string[], exclude?: string[]) {
    const isMatch = (pattern?:string[]) => {
        if (Array.isArray(pattern)) {
            if(pattern.length === 0){
                return undefined;
            }
            return pattern.includes(key);
        }
    };
    const includeIsMatch = isMatch(include);
    if (typeof includeIsMatch === 'undefined') {
        const excludeIsMatch = isMatch(exclude);
        return typeof excludeIsMatch === 'undefined' ? true : !excludeIsMatch;
    }
    return includeIsMatch;
}

/**
 * 处理一组 Promise 不管成功失败 都返回数据
 * @param allPromises
 */
function allSettled(allPromises: Promise<GenMessage>[]): Promise<GenMessageWrapper> {
    const successMessages: GenMessage[] = [];
    const errorMessages: GenMessage[] = [];
    let count = 0;
    return new Promise(resolve => {
        allPromises.forEach(item => {
            item.then(successMessage => {
                successMessages.push(successMessage);
            })
                .catch(errMessage => {
                    errorMessages.push(errMessage);
                })
                .finally(() => {
                    count++;
                    if (count === allPromises.length) {
                        resolve({
                            successMessages,
                            errorMessages,
                        });
                    }
                });
        });
    });
}

function genFile(filename: string, content: string) {
    const fileBasename = basename(filename);
    return new Promise<GenMessage>((resolve, reject) => {
        writeFile(filename, content)
            .then(() => {
                resolve(getSuccessMessage(filename));
            })
            .catch(err => {
                reject(getErrorMessage(filename));
                return console.error(`Failed to store ${fileBasename}:${err.message}`);
            });
    });
}

function getSuccessMessage(outputPath: string): GenMessage {
    return {
        outputPath,
        message: `${outputPath} generation succeeded`,
        status: Status.success,
    };
}

function getErrorMessage(outputPath: string): GenMessage {
    return {
        outputPath,
        message: `${outputPath} generation failed`,
        status: Status.error,
    };
}
