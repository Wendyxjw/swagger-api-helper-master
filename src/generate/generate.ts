import { Options } from './interfaces';
import ApiGenerator from './ApiGenerator';

const generate = async (options: Options) => {
    const apiGenerator = new ApiGenerator(options);
    await apiGenerator.fetch();
    return apiGenerator.generate();
}; 

export default generate;
