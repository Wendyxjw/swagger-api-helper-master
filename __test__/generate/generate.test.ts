import path from 'path';
import fs from 'fs';
import generate from '../../src/generate';
import { Authorization, urls } from '../config';
import { removeFolder } from '../utils';

it('generate services', async () => {
    const outputPath = path.join(__dirname, './services');
    removeFolder(outputPath);

    await generate({
        tagAlias: {
            pet: 'petAlias',
        },
        fetchOptions: {
            headers: {
                Authorization,
            },
        },
        urls: [urls[0]],
        outputPath,
        hasExtraFetchOptions: true,
    });

    await generate({
        tagAlias: {
            pet: 'petAlias',
        },
        fetchOptions: {
            headers: {
                Authorization,
            },
        },
        urls: [urls[1]],
        include: ['user'],
        outputPath,
        hasExtraFetchOptions: true,
    });

    const dirs = fs.readdirSync(outputPath);

    dirs.forEach(dir => {
        const filenames = fs.readdirSync(path.join(outputPath, dir));
        filenames.forEach(filename => {
            const content = fs.readFileSync(path.join(outputPath, dir, filename), {
                encoding: 'utf-8',
            });
            expect(content).toMatchSnapshot();
        });
    });
}, 10000);
