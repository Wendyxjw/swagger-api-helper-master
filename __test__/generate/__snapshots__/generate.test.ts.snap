// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate services 1`] = `
"export interface ApiResponse {
	code?: number;
	type?: string;
	message?: string;
}
export interface Category {
	id?: number;
	name?: string;
}
export interface Pet {
	id?: number;
	category?: Category;
	name: string; // example: \\"doggie\\"
	photoUrls: Array<string>;
	tags?: Array<Tag>;
	status?: 'available' | 'pending' | 'sold'; // pet status in the store
}
export interface Tag {
	id?: number;
	name?: string;
}
export interface Order {
	id?: number;
	petId?: number;
	quantity?: number;
	shipDate?: string;
	status?: 'placed' | 'approved' | 'delivered'; // Order Status
	complete?: boolean;
}
export interface User {
	id?: number;
	username?: string;
	firstName?: string;
	lastName?: string;
	email?: string;
	password?: string;
	phone?: string;
	userStatus?: number; // User Status
}"
`;

exports[`generate services 2`] = `
"import stringify from '@/utils/stringify';
import request from '@/utils/request';
import { ExtraFetchOptions } from '@/types';
import { ApiResponse, Pet } from './interfaces';


export interface PostPetPetIdUploadImagePayload extends ExtraFetchOptions {
	petId: number;
	/**
	 * 
	 * export interface FormContent {
	 * 	additionalMetadata?: string; // Additional data to pass to server
	 * 	file?: File; // file to upload
	 * }
	 * 
	 */
	body: FormData;
}
    
/**
 * @summary uploads an image
 */
export async function postPetPetIdUploadImage(payload: PostPetPetIdUploadImagePayload) {
	const { petId, ...extraFetchOptions } = payload;
    return request<ApiResponse>(\`/v2/pet/\${petId}/uploadImage\`, {
		...extraFetchOptions,
        method: 'post',
    });
}

export interface PostPetPayload extends ExtraFetchOptions {
	/**
	 * Pet object that needs to be added to the store
	 */
	body: Pet;
}
    
/**
 * @summary Add a new pet to the store
 */
export async function postPet(payload: PostPetPayload) {
	const { body, ...extraFetchOptions } = payload;
    return request<undefined>(\`/v2/pet\`, {
		...extraFetchOptions,
        method: 'post',
		body,
    });
}

export interface PutPetPayload extends ExtraFetchOptions {
	/**
	 * Pet object that needs to be added to the store
	 */
	body: Pet;
}
    
/**
 * @summary Update an existing pet
 */
export async function putPet(payload: PutPetPayload) {
	const { body, ...extraFetchOptions } = payload;
    return request<undefined>(\`/v2/pet\`, {
		...extraFetchOptions,
        method: 'put',
		body,
    });
}


export interface GetPetFindByStatusQuery {
	status: Array<'available' | 'pending' | 'sold'>; // Status values that need to be considered for filter
}

export interface GetPetFindByStatusPayload extends ExtraFetchOptions {
	query: GetPetFindByStatusQuery;
}
    
/**
 * @summary Finds Pets by status
 * @description Multiple status values can be provided with comma separated strings
 */
export async function getPetFindByStatus(payload: GetPetFindByStatusPayload) {
	const { query, ...extraFetchOptions } = payload;
    return request<Pet[]>(\`/v2/pet/findByStatus?\${stringify(query)}\`, {
		...extraFetchOptions,
        method: 'get',
    });
}


export interface GetPetFindByTagsQuery {
	tags: Array<string>; // Tags to filter by
}

export interface GetPetFindByTagsPayload extends ExtraFetchOptions {
	query: GetPetFindByTagsQuery;
}
    
/**
 * 废弃不用 
 * @deprecated
 * @summary Finds Pets by tags
 * @description Multiple tags can be provided with comma separated strings. Use tag1, tag2, tag3 for testing.
 */
export async function getPetFindByTags(payload: GetPetFindByTagsPayload) {
	const { query, ...extraFetchOptions } = payload;
    return request<Pet[]>(\`/v2/pet/findByTags?\${stringify(query)}\`, {
		...extraFetchOptions,
        method: 'get',
    });
}

export interface GetPetPetIdPayload extends ExtraFetchOptions {
	petId: number;
}
    
/**
 * @summary Find pet by ID
 * @description Returns a single pet
 */
export async function getPetPetId(payload: GetPetPetIdPayload) {
	const { petId, ...extraFetchOptions } = payload;
    return request<Pet>(\`/v2/pet/\${petId}\`, {
		...extraFetchOptions,
        method: 'get',
    });
}

export interface PostPetPetIdPayload extends ExtraFetchOptions {
	petId: number;
	/**
	 * 
	 * export interface FormContent {
	 * 	name?: string; // Updated name of the pet
	 * 	status?: string; // Updated status of the pet
	 * }
	 * 
	 */
	body: FormData;
}
    
/**
 * @summary Updates a pet in the store with form data
 */
export async function postPetPetId(payload: PostPetPetIdPayload) {
	const { petId, ...extraFetchOptions } = payload;
    return request<undefined>(\`/v2/pet/\${petId}\`, {
		...extraFetchOptions,
        method: 'post',
    });
}

export interface DeletePetPetIdPayload extends ExtraFetchOptions {
	petId: number;
}
    
/**
 * @summary Deletes a pet
 */
export async function deletePetPetId(payload: DeletePetPetIdPayload) {
	const { petId, ...extraFetchOptions } = payload;
    return request<undefined>(\`/v2/pet/\${petId}\`, {
		...extraFetchOptions,
        method: 'delete',
    });
}


"
`;

exports[`generate services 3`] = `
"import request from '@/utils/request';
import { ExtraFetchOptions } from '@/types';
import { Order } from './interfaces';


export interface PostStoreOrderPayload extends ExtraFetchOptions {
	/**
	 * order placed for purchasing the pet
	 */
	body: Order;
}
    
/**
 * @summary Place an order for a pet
 */
export async function postStoreOrder(payload: PostStoreOrderPayload) {
	const { body, ...extraFetchOptions } = payload;
    return request<Order>(\`/v2/store/order\`, {
		...extraFetchOptions,
        method: 'post',
		body,
    });
}

export interface GetStoreOrderOrderIdPayload extends ExtraFetchOptions {
	orderId: number;
}
    
/**
 * @summary Find purchase order by ID
 * @description For valid response try integer IDs with value >= 1 and <= 10. Other values will generated exceptions
 */
export async function getStoreOrderOrderId(payload: GetStoreOrderOrderIdPayload) {
	const { orderId, ...extraFetchOptions } = payload;
    return request<Order>(\`/v2/store/order/\${orderId}\`, {
		...extraFetchOptions,
        method: 'get',
    });
}

export interface DeleteStoreOrderOrderIdPayload extends ExtraFetchOptions {
	orderId: number;
}
    
/**
 * @summary Delete purchase order by ID
 * @description For valid response try integer IDs with positive integer value. Negative or non-integer values will generate API errors
 */
export async function deleteStoreOrderOrderId(payload: DeleteStoreOrderOrderIdPayload) {
	const { orderId, ...extraFetchOptions } = payload;
    return request<undefined>(\`/v2/store/order/\${orderId}\`, {
		...extraFetchOptions,
        method: 'delete',
    });
}

export interface GetStoreInventoryPayload extends ExtraFetchOptions {

}
    
/**
 * @summary Returns pet inventories by status
 * @description Returns a map of status codes to quantities
 */
export async function getStoreInventory(payload: GetStoreInventoryPayload) {
	const extraFetchOptions = payload;
    return request<object>(\`/v2/store/inventory\`, {
		...extraFetchOptions,
        method: 'get',
    });
}


"
`;

exports[`generate services 4`] = `
"import stringify from '@/utils/stringify';
import request from '@/utils/request';
import { ExtraFetchOptions } from '@/types';
import { User } from './interfaces';


export interface PostUserCreateWithArrayPayload extends ExtraFetchOptions {
	/**
	 * List of user object
	 */
	body: User[];
}
    
/**
 * @summary Creates list of users with given input array
 */
export async function postUserCreateWithArray(payload: PostUserCreateWithArrayPayload) {
	const { body, ...extraFetchOptions } = payload;
    return request<undefined>(\`/v2/user/createWithArray\`, {
		...extraFetchOptions,
        method: 'post',
		body,
    });
}

export interface PostUserCreateWithListPayload extends ExtraFetchOptions {
	/**
	 * List of user object
	 */
	body: User[];
}
    
/**
 * @summary Creates list of users with given input array
 */
export async function postUserCreateWithList(payload: PostUserCreateWithListPayload) {
	const { body, ...extraFetchOptions } = payload;
    return request<undefined>(\`/v2/user/createWithList\`, {
		...extraFetchOptions,
        method: 'post',
		body,
    });
}

export interface GetUserUsernamePayload extends ExtraFetchOptions {
	username: string;
}
    
/**
 * @summary Get user by user name
 */
export async function getUserUsername(payload: GetUserUsernamePayload) {
	const { username, ...extraFetchOptions } = payload;
    return request<User>(\`/v2/user/\${username}\`, {
		...extraFetchOptions,
        method: 'get',
    });
}

export interface PutUserUsernamePayload extends ExtraFetchOptions {
	username: string;
	/**
	 * Updated user object
	 */
	body: User;
}
    
/**
 * @summary Updated user
 * @description This can only be done by the logged in user.
 */
export async function putUserUsername(payload: PutUserUsernamePayload) {
	const { username, body, ...extraFetchOptions } = payload;
    return request<undefined>(\`/v2/user/\${username}\`, {
		...extraFetchOptions,
        method: 'put',
		body,
    });
}

export interface DeleteUserUsernamePayload extends ExtraFetchOptions {
	username: string;
}
    
/**
 * @summary Delete user
 * @description This can only be done by the logged in user.
 */
export async function deleteUserUsername(payload: DeleteUserUsernamePayload) {
	const { username, ...extraFetchOptions } = payload;
    return request<undefined>(\`/v2/user/\${username}\`, {
		...extraFetchOptions,
        method: 'delete',
    });
}


export interface GetUserLoginQuery {
	username: string; // The user name for login
	password: string; // The password for login in clear text
}

export interface GetUserLoginPayload extends ExtraFetchOptions {
	query: GetUserLoginQuery;
}
    
/**
 * @summary Logs user into the system
 */
export async function getUserLogin(payload: GetUserLoginPayload) {
	const { query, ...extraFetchOptions } = payload;
    return request<string>(\`/v2/user/login?\${stringify(query)}\`, {
		...extraFetchOptions,
        method: 'get',
    });
}

export interface GetUserLogoutPayload extends ExtraFetchOptions {

}
    
/**
 * @summary Logs out current logged in user session
 */
export async function getUserLogout(payload: GetUserLogoutPayload) {
	const extraFetchOptions = payload;
    return request<undefined>(\`/v2/user/logout\`, {
		...extraFetchOptions,
        method: 'get',
    });
}

export interface PostUserPayload extends ExtraFetchOptions {
	/**
	 * Created user object
	 */
	body: User;
}
    
/**
 * @summary Create user
 * @description This can only be done by the logged in user.
 */
export async function postUser(payload: PostUserPayload) {
	const { body, ...extraFetchOptions } = payload;
    return request<undefined>(\`/v2/user\`, {
		...extraFetchOptions,
        method: 'post',
		body,
    });
}


"
`;

exports[`generate services 5`] = `
"export interface User {
	id?: number;
	username?: string;
	firstName?: string;
	lastName?: string;
	email?: string;
	password?: string;
	phone?: string;
	userStatus?: number; // User Status
}"
`;

exports[`generate services 6`] = `
"import stringify from '@/utils/stringify';
import request from '@/utils/request';
import { ExtraFetchOptions } from '@/types';
import { User } from './interfaces';


export interface PostUserCreateWithArrayPayload extends ExtraFetchOptions {
	/**
	 * List of user object
	 */
	body: User[];
}
    
/**
 * @summary Creates list of users with given input array
 */
export async function postUserCreateWithArray(payload: PostUserCreateWithArrayPayload) {
	const { body, ...extraFetchOptions } = payload;
    return request<undefined>(\`/v2/user/createWithArray\`, {
		...extraFetchOptions,
        method: 'post',
		body,
    });
}

export interface PostUserCreateWithListPayload extends ExtraFetchOptions {
	/**
	 * List of user object
	 */
	body: User[];
}
    
/**
 * @summary Creates list of users with given input array
 */
export async function postUserCreateWithList(payload: PostUserCreateWithListPayload) {
	const { body, ...extraFetchOptions } = payload;
    return request<undefined>(\`/v2/user/createWithList\`, {
		...extraFetchOptions,
        method: 'post',
		body,
    });
}

export interface GetUserUsernamePayload extends ExtraFetchOptions {
	username: string;
}
    
/**
 * @summary Get user by user name
 */
export async function getUserUsername(payload: GetUserUsernamePayload) {
	const { username, ...extraFetchOptions } = payload;
    return request<User>(\`/v2/user/\${username}\`, {
		...extraFetchOptions,
        method: 'get',
    });
}

export interface PutUserUsernamePayload extends ExtraFetchOptions {
	username: string;
	/**
	 * Updated user object
	 */
	body: User;
}
    
/**
 * @summary Updated user
 * @description This can only be done by the logged in user.
 */
export async function putUserUsername(payload: PutUserUsernamePayload) {
	const { username, body, ...extraFetchOptions } = payload;
    return request<undefined>(\`/v2/user/\${username}\`, {
		...extraFetchOptions,
        method: 'put',
		body,
    });
}

export interface DeleteUserUsernamePayload extends ExtraFetchOptions {
	username: string;
}
    
/**
 * @summary Delete user
 * @description This can only be done by the logged in user.
 */
export async function deleteUserUsername(payload: DeleteUserUsernamePayload) {
	const { username, ...extraFetchOptions } = payload;
    return request<undefined>(\`/v2/user/\${username}\`, {
		...extraFetchOptions,
        method: 'delete',
    });
}


export interface GetUserLoginQuery {
	username: string; // The user name for login
	password: string; // The password for login in clear text
}

export interface GetUserLoginPayload extends ExtraFetchOptions {
	query: GetUserLoginQuery;
}
    
/**
 * @summary Logs user into the system
 */
export async function getUserLogin(payload: GetUserLoginPayload) {
	const { query, ...extraFetchOptions } = payload;
    return request<string>(\`/v2/user/login?\${stringify(query)}\`, {
		...extraFetchOptions,
        method: 'get',
    });
}

export interface GetUserLogoutPayload extends ExtraFetchOptions {

}
    
/**
 * @summary Logs out current logged in user session
 */
export async function getUserLogout(payload: GetUserLogoutPayload) {
	const extraFetchOptions = payload;
    return request<undefined>(\`/v2/user/logout\`, {
		...extraFetchOptions,
        method: 'get',
    });
}

export interface PostUserPayload extends ExtraFetchOptions {
	/**
	 * Created user object
	 */
	body: User;
}
    
/**
 * @summary Create user
 * @description This can only be done by the logged in user.
 */
export async function postUser(payload: PostUserPayload) {
	const { body, ...extraFetchOptions } = payload;
    return request<undefined>(\`/v2/user\`, {
		...extraFetchOptions,
        method: 'post',
		body,
    });
}


"
`;
